simpleast-core/src/main/java/com/discord/simpleast/core/parser/Parser.kt

```diff
@@ -92,7 +96,7 @@ open class Parser<R, T : Node<R>> @JvmOverloads constructor(private val enableDe
       }
 
       if (!foundRule) {
-        throw RuntimeException("failed to find rule to match source: \"$inspectionSource\"")
+        throw ParseException("failed to find rule to match source", source)
```

Comment:
1. docs need to be updated to say this is being thrown.
2. I would also update the sample/test code to include handling for this error.
3. Update README?

Reply:
I don't wanna add it to the README given that this isn't an expected case for users to handle, but I can at least mention it in the doc and have a test.

Reply:
actually I can't test for it... since we don't know the cause...